lib/hashes.md

```diff
@@ -5,40 +5,42 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
+organize a count of like items, or categorize items into desired groups, etc.  Hashes have much more variety in terms of what they can offer.
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
-
+A particular value from a hash is retrieved by using the key for that value.  For instance, if this is our hash:  grades = { "Mike Ducey" => A, "Pat Ducey" => B+ }, you reference the hash like an array, but instead of specifying the number of the item, you use the key.  So, in this case, if I write grades["Mike Ducey"], I shall get => A as a response.
 # How do you add information to a Hash?
 
-Replace me with your response...
-
+This is done by using the "bracket-equals method" (https://github.com/learn-co-curriculum/hash-overview-readme).  That is, if you have a hash and you want another key in the hash, you can write the name of the key in the brackets and the value after the equal sign.  I'll add a grade to my grades hash: grades["Kevin Ducey"] = A-
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+This is achieved by using the each function.  If you want to change each value, you use something of the form: grades.each do |x| where x is the change to be done using code.
 
 # How would you change the value of a particular element in a Hash?
 
-Replace me with your response...
+This is done via utilizing the same sort of method as assigning a new key/value pair.  So, it is basically an overwriting of a value.  I could write grades("Pat Ducey") = C+ and that would be the new value for the key "Pat Ducey"
 
 # How do you delete an element from a Hash?
 
-Replace me with your response...
+To delete a particular element from a hash, a user will use the delete function and put the key that identifies the element in parentheses.  For instance, if I were to write grades.delete("Mike Ducey"), it would remove the key-value pair of "Mike Ducey" +> A from the grades hash.
 
 # What happens if you try to retrieve an element from a Hash that does not exist in the Hash?
 
-Replace me with your response...
+If you use a key that has no corresponding element, it will return the default value.   The default can be changed to represent what the default value is.  So, in our grades hash, we could write grades.default = "That student does not exist"  So if I went to call grades["David Bowie"] from the grades hash, it would return That student does not exist.
 
 # How do you determine how many elements are in a Hash?
 
-Replace me with your response...
\ No newline at end of file
+The length function would give the number of key-value pairs (and then supposedly the number of elements!) I'm a bit vague on this, but I think from what I've learned that you can use the length_value 
```

Comment:
> The length function would give the number of key-value pairs (and then supposedly the number of elements!)

Can you say more what you mean by "the number of elements"?

> I'm a bit vague on this, but I think from what I've learned that you can use the length_value 

I'm not sure I understand. Is `length_value` a method you an call on a Hash?